mysensors/sensor.py

```
@@ -103,46 +108,124 @@ def add_child_sensor(self, child_id, child_type, description=""):
         self.children[child_id] = ChildSensor(child_id, child_type, description)
         return child_id
 
-    def set_child_value(self, child_id, value_type, value, **kwargs):
-        """Set a child sensor's value."""
-        children = kwargs.get("children", self.children)
-        if not isinstance(children, dict) or child_id not in children:
+    def get_desired_value(self, child_id, value_type):
+        """Return sensor state value taking into account node type."""
+        if child_id not in self.children:
             return None
-        msg_type = kwargs.get("msg_type", 1)
-        ack = kwargs.get("ack", 0)
-        msg = Message().modify(
+
+        value = None
+
+        if self.is_smart_sleep_node:
+            child = self.new_state[child_id]
+            value = child.values.get(value_type) if child else None
+
+        if value:
+            return value
+
+        child = self.children[child_id]
+
+        return child.values.get(value_type)
+
+    def init_smart_sleep_mode(self):
+        """Init desired state dict for all known children."""
+        for child in self.children.values():
+            if child.id in self.new_state:
+                continue
+
+            self.new_state[child.id] = ChildSensor(
+                child.id, child.type, child.description
+            )
+
+    def set_child_desired_state(self, child_id, value_type, value):
+        """Set a desired child sensor's value for smart sleep nodes."""
+        if child_id not in self.new_state:
+            _LOGGER.warning(
+                "Warning: attempt to set a desired state value"
+                " on non-smart sleep node: node %s, child %s, type %s, value %s",
+                self.sensor_id,
+                child_id,
+                value_type,
+                value,
+            )
+            return
+
+        if not self.validate_child_state(child_id, value_type, value):
```

This should also probably raise an error. It's a developer mistake.